Index: qpdf-4.0.0/qpdf/test_driver.cc
===================================================================
--- qpdf-4.0.0.orig/qpdf/test_driver.cc	2012-12-31 11:43:27.000000000 -0500
+++ qpdf-4.0.0/qpdf/test_driver.cc	2013-01-16 13:42:51.089682329 -0500
@@ -1188,7 +1188,7 @@
             bool is_binary = false;
             for (size_t i = 0; i < data.size(); ++i)
             {
-                if (data[i] < 0)
+                if ((data[i] < 0) || (data[i] > 126))
                 {
                     is_binary = true;
                     break;
