Index: qpdf-2.3.1/libqpdf/PCRE.cc
===================================================================
--- qpdf-2.3.1.orig/libqpdf/PCRE.cc	2011-12-28 17:19:40.000000000 -0500
+++ qpdf-2.3.1/libqpdf/PCRE.cc	2012-04-06 21:42:59.041277541 -0400
@@ -144,7 +144,7 @@
     this->code = pcre_compile(pattern, options, &errptr, &erroffset, 0);
     if (this->code)
     {
-	this->nbackrefs = pcre_info(this->code, 0, 0);
+	pcre_fullinfo(this->code, 0, PCRE_INFO_CAPTURECOUNT, &(this->nbackrefs));
     }
     else
     {
