Index: qpdf-2.3.1/include/qpdf/QPDFExc.hh
===================================================================
--- qpdf-2.3.1.orig/include/qpdf/QPDFExc.hh	2011-12-28 17:19:40.000000000 -0500
+++ qpdf-2.3.1/include/qpdf/QPDFExc.hh	2012-05-19 09:19:22.148246280 -0400
@@ -8,6 +8,7 @@
 #ifndef __QPDFEXC_HH__
 #define __QPDFEXC_HH__
 
+#include <sys/types.h>          // off_t - not for upstream
 #include <qpdf/DLL.h>
 #include <qpdf/Constants.h>
 #include <stdexcept>
Index: qpdf-2.3.1/include/qpdf/QPDFObjectHandle.hh
===================================================================
--- qpdf-2.3.1.orig/include/qpdf/QPDFObjectHandle.hh	2011-12-28 17:19:40.000000000 -0500
+++ qpdf-2.3.1/include/qpdf/QPDFObjectHandle.hh	2012-05-19 09:19:22.148246280 -0400
@@ -8,6 +8,7 @@
 #ifndef __QPDFOBJECTHANDLE_HH__
 #define __QPDFOBJECTHANDLE_HH__
 
+#include <sys/types.h>          // off_t - not for upstream
 #include <string>
 #include <vector>
 #include <set>
Index: qpdf-2.3.1/include/qpdf/QPDF.hh
===================================================================
--- qpdf-2.3.1.orig/include/qpdf/QPDF.hh	2011-12-28 17:19:40.000000000 -0500
+++ qpdf-2.3.1/include/qpdf/QPDF.hh	2012-05-19 09:19:22.148246280 -0400
@@ -8,6 +8,7 @@
 #ifndef __QPDF_HH__
 #define __QPDF_HH__
 
+#include <sys/types.h>          // off_t - not for upstream
 #include <stdio.h>
 #include <string>
 #include <map>
Index: qpdf-2.3.1/examples/pdf-mod-info.cc
===================================================================
--- qpdf-2.3.1.orig/examples/pdf-mod-info.cc	2011-12-28 17:19:40.000000000 -0500
+++ qpdf-2.3.1/examples/pdf-mod-info.cc	2012-05-19 09:20:18.016247014 -0400
@@ -8,6 +8,7 @@
 #include <iostream>
 #include <string.h>
 #include <stdlib.h>
+#include <unistd.h>              // unlink - not for upstream
 
 static char const* version = "1.1";
 static char const* whoami = 0;
